start-site/src/main/java/io/spring/start/site/extension/springcloud/SpringCloudFunctionHelpDocumentCustomizer.java

```diff
@@ -0,0 +1,170 @@
+import io.spring.initializr.generator.buildsystem.maven.MavenBuildSystem;
+import io.spring.initializr.generator.io.template.MustacheTemplateRenderer;
+import io.spring.initializr.generator.project.ResolvedProjectDescription;
+import io.spring.initializr.generator.spring.documentation.HelpDocument;
+import io.spring.initializr.generator.spring.documentation.HelpDocumentCustomizer;
+import io.spring.initializr.generator.version.Version;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * A {@link HelpDocumentCustomizer} that adds information regarding build file setup for
+ * running Spring Cloud Function projects on various cloud platforms.
+ *
+ * @author Olga Maciaszek-Sharma
+ */
+class SpringCloudFunctionHelpDocumentCustomizer implements HelpDocumentCustomizer {
+
+	private static final Log LOG = LogFactory.getLog(SpringCloudFunctionHelpDocumentCustomizer.class);
+
+	private static final String SNAPSHOT = "SNAPSHOT";
+
+	private static final String MISSING_TEMPLATE_SUFFIX = "missing";
+
+	private static final String SPRING_CLOUD_FUNCTION_DEPENDENCY_ID = "cloud-function";
+
+	private static final String TEMPLATE_PREFIX = "spring-cloud-function-build-setup-";
+
+	private static final String SPRING_CLOUD_FUNCTION_ARTIFACT_ID = "org.springframework.cloud:spring-cloud-function-core";
+
+	private final Set<String> buildDependencies;
+
+	private final MustacheTemplateRenderer templateRenderer;
+
+	private final SpringCloudProjectVersionResolver projectVersionResolver;
+
+	private final ResolvedProjectDescription description;
+
+	SpringCloudFunctionHelpDocumentCustomizer(Build build, ResolvedProjectDescription description,
+			MustacheTemplateRenderer templateRenderer, SpringCloudProjectVersionResolver projectVersionResolver) {
+		this.buildDependencies = build.dependencies().ids().collect(Collectors.toSet());
+		this.description = description;
+		this.templateRenderer = templateRenderer;
+		this.projectVersionResolver = projectVersionResolver;
+	}
+
+	@Override
+	public void customize(HelpDocument helpDocument) {
+		this.buildDependencies.stream()
+				.filter((dependencyId) -> dependencyId.equals(SPRING_CLOUD_FUNCTION_DEPENDENCY_ID)).findAny()
+				.ifPresent((dependency) -> addBuildSetupInfo(helpDocument));
+	}
+
+	private void addBuildSetupInfo(HelpDocument helpDocument) {
+		Version bootVersion = this.description.getPlatformVersion();
+		String springCloudFunctionVersion = this.projectVersionResolver.resolveVersion(bootVersion,
+				SPRING_CLOUD_FUNCTION_ARTIFACT_ID);
+		if (springCloudFunctionVersion == null) {
+			LOG.warn("Spring Cloud Function version could not be resolved for Spring Boot version: "
+					+ bootVersion.toString());
+			return;
+		}
+		if (isSnapshot(springCloudFunctionVersion)) {
+			LOG.debug("Spring Cloud Function version " + springCloudFunctionVersion
+					+ " is a snapshot. No documents are present for this version to link to.");
```

Comment:
Not sure I followed why the document wouldn't be there for snapshots.

Reply:
The docs are not being published for snapshots, i.e. this link can be accessed online: https://cloud.spring.io/spring-cloud-static/spring-cloud-function/3.0.0.M1/spring-cloud-function.html , but there is no: https://cloud.spring.io/spring-cloud-static/spring-cloud-function/3.0.0.BUILD-SNAPSHOT/spring-cloud-function.html